examples/miscellaneous/plot_pipeline_display.py

```diff
@@ -0,0 +1,218 @@
+# preprocessing steps, :class:`~sklearn.preprocessing.PolynomialFeatures` and
+# :class:`~sklearn.preprocessing.StandardScaler`, and a classifer step,
+# :class:`~sklearn.linear_model.LogisticRegression`, and displays its visual
+# representation.
+
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler, PolynomialFeatures
+from sklearn.linear_model import LogisticRegression
+from sklearn import set_config
+
+steps = [
+    ("polynomial", PolynomialFeatures()),
+    ("standard_scaler", StandardScaler()),
+    ("classifier", LogisticRegression()),
+]
+pipe = Pipeline(steps)
+
+# %%
+# To visualize the diagram, change to display='diagram'
+set_config(display="diagram")
+pipe  # click on the diagram below to see the details of each step
+
+# %%
+# Displaying a Pipeline and Dimensionality Reduction and Classifier
+################################################################################
+# This section constructs a :class:`~sklearn.pipeline.Pipeline` with a
+# dimensionality reduction step, :class:`~sklearn.decomposition.PCA`,
+# a classifier, :class:`~sklearn.svm.SVC`, and displays its visual
+# representation.
+
+from sklearn.pipeline import Pipeline
+from sklearn.svm import SVC
+from sklearn.decomposition import PCA
+from sklearn import set_config
+
+steps = [("reduce_dim", PCA()), ("classifier", SVC())]
+pipe = Pipeline(steps)
+
+# %%
+# To visualize the diagram, change to `display='diagram'`
+set_config(display="diagram")
+pipe  # click on the diagram below to see the details of each step
+
+# %%
+# Displaying a Complex Pipeline Chaining a Column Transformer
+################################################################################
+# This section constructs a complex :class:`~sklearn.pipeline.Pipeline` with a
+# :class:`~sklearn.compose.ColumnTransformer` and a classifier,
+# :class:`~sklearn.linear_model.LogisticRegression`, and displays its visual
+# representation.
+
+from sklearn.compose import make_column_selector as selector
+from sklearn.preprocessing import OneHotEncoder, StandardScaler
+from sklearn.compose import ColumnTransformer
+from sklearn.linear_model import LogisticRegression
+from sklearn.pipeline import make_pipeline
+import pandas as pd
+from sklearn import set_config
+
+numerical_columns_selector = selector(dtype_exclude=object)
+categorical_columns_selector = selector(dtype_include=object)
+
+data = pd.DataFrame()
+
+numerical_columns = numerical_columns_selector(data)
+categorical_columns = categorical_columns_selector(data)
+
+categorical_preprocessor = OneHotEncoder(handle_unknown="ignore")
+numerical_preprocessor = StandardScaler()
+
+preprocessor = ColumnTransformer(
+    [
+        ("one-hot-encoder", categorical_preprocessor, categorical_columns),
+        ("standard-scaler", numerical_preprocessor, numerical_columns),
+    ]
+)
+
+pipe = make_pipeline(preprocessor, LogisticRegression(max_iter=500))
```

Comment:
I think we can keep it simple here and explicitly pass in the column names. This keeps the focus on the visualization. In this case we can use pipelines:

```py
from sklearn import set_config

numeric_preprocessor = Pipeline(
    steps=[
        ("imputation_mean", SimpleImputer(missing_values=np.nan, strategy="mean")),
        ("scaler", StandardScaler()),
    ]
)

categorical_preprocessor = Pipeline(
    steps=[
        (
            "imputation_constant",
            SimpleImputer(fill_value="missing", strategy="constant"),
        ),
        ("onehot", OneHotEncoder(handle_unknown="ignore")),
    ]
)

preprocessor = ColumnTransformer(
    [
        ("categorical", categorical_preprocessor, ["state", "gender"]),
        ("numerical", numeric_preprocessor, ["age", "weight"]),
    ]
)

pipe = make_pipeline(preprocessor, LogisticRegression(max_iter=500))
```

Reply:
updated